kube-core/src/schema.rs

```diff
@@ -36,6 +37,110 @@ impl Visitor for StructuralSchemaRewriter {
+                //                 }
+                //             },
+                //             "required": ["int"],
+                //         },
+                //         "VariantTwo": {
+                //             "type": "object",
+                //         }
+                //     },
+                //     "oneOf": [
+                //     {
+                //         "required": ["variantOne"],
+                //         "description": "First variant with an int"
+                //     },
+                //     {
+                //         "required": ["variantTwo"],
+                //         "description": "Second variant which doesn't has an attribute"
+                //     }
+                //     ],
+                //     "description": "This is a complex enum"
+                // }
+                //
+                // The correct solution requires to move the description from the oneOf section to to the complexEnum.properties
+                // The corrected solution looks like follows
+                // "complexEnum": {
+                //     "type": "object",
+                //     "properties": {
+                //         "variantOne": {
+                //             "type": "object",
+                //             "properties": {
+                //                 "int": {
+                //                     "type": "integer",
+                //                     "format": "int32"
+                //                 }
+                //             },
+                //             "required": ["int"],
+                //             "description": "First variant with an int"
+                //         },
+                //         "variantTwo": {
+                //             "type": "object",
+                //             "description": "Second variant which doesn't has an attribute"
+                //         }
+                //     },
+                //     "oneOf": [
+                //     {
+                //         "required": ["variantOne"],
+                //     },
+                //     {
+                //         "required": ["variantTwo"],
+                //     }
+                //     ],
+                //     "description": "This is a complex enum"
+                // }
+                //
+                // The following lines move the descriptions to the correct places
+                if let Schema::Object(SchemaObject {
+                    metadata: Some(variant_metadata),
+                    object: Some(variant_obj),
+                    ..
+                }) = variant
+                {
+                    if variant_obj.properties.len() == 1 {
+                        if let Some(description) = std::mem::take(&mut variant_metadata.description) {
+                            if let Some(Schema::Object(variant_object)) =
+                                variant_obj.properties.values_mut().next()
```

Comment:
It'd be nice if we could throw a warning somehow if there isn't exactly one item... Might be worth defining something like:

```rust
fn only_item<I: Iterator>(i: I) -> Option<I::Item> {
    let item = i.next()?;
    if i.next().is_some() {
        return None;
    }
    Some(item)
}
```